src/charm_state.py

```diff
@@ -309,6 +309,9 @@ def from_charm(cls, charm: CharmBase) -> "State":
         # Convert pydantic object to python object serializable by json module.
         state_dict["proxy_config"] = json.loads(state_dict["proxy_config"].json())
         state_dict["charm_config"] = json.loads(state_dict["charm_config"].json())
+        state_dict["ssh_debug_info"] = (
+            json.loads(state_dict["ssh_debug_info"].json()) if ssh_debug_info else None
```

Comment:
Did you mean `if ssh_debug_info in state_dict else None`?

Reply:
`ssh_debug_info` is None if it's not available, so either value can be used here :D